src/StateManager.ts

```diff
@@ -589,220 +589,230 @@ export default class StateManager {
-              StateManager._nodeWrappers.push(newNode);
-              StateManager._nodeLayer.add(newNode.nodeGroup);
-      
-              data.nodes.push(newNode);
-              nodeMap.set(obj.id, newNode);
-            }
-          });
-      
-          // Then, create new transitions, using the new nodes
-          this._clipboard.forEach((obj) => {
-            if (obj instanceof TransitionWrapper) {
-              const sourceNode = nodeMap.get(obj.sourceNode.id);
-              const destNode = nodeMap.get(obj.destNode.id);
-      
-              // Only create transitions if both source and dest nodes were copied
-              if (sourceNode && destNode) {
-                const newTransition = new TransitionWrapper(
-                  sourceNode,
-                  destNode,
-                  obj.isEpsilonTransition,
-                  new Set(obj.tokens) // Copy tokens
-                );
-      
-                StateManager._transitionWrappers.push(newTransition);
-                StateManager._transitionLayer.add(newTransition.konvaGroup);
-      
-                data.transitions.push(newTransition);
-              }
-            }
-          });
-        } else {
-          // Redoing the paste action, reuse nodes and transitions
-          data.nodes.forEach((node) => {
-            StateManager._nodeWrappers.push(node);
-            StateManager._nodeLayer.add(node.nodeGroup);
-          });
-      
-          data.transitions.forEach((transition) => {
-            StateManager._transitionWrappers.push(transition);
-            StateManager._transitionLayer.add(transition.konvaGroup);
-          });
-        }
-      
-        StateManager._nodeLayer?.draw();
-        StateManager._transitionLayer?.draw();
-      
-        StateManager.updateTransitions();
-      
-        StateManager.deselectAllObjects();
-        data.nodes.forEach((node) => StateManager.selectObject(node));
-      };
-    
-      let performPasteBackward = (data: PasteActionData) => {
-
-        StateManager.deselectAllObjects();
-    
-        // Remove transitions
-        data.transitions.forEach((transition) => {
-          StateManager._transitionWrappers = StateManager._transitionWrappers.filter(t => t !== transition);
-          transition.konvaGroup.remove();
-        });
-    
-        // Remove nodes
-        data.nodes.forEach((node) => {
-          StateManager._nodeWrappers = StateManager._nodeWrappers.filter(n => n !== node);
-          node.nodeGroup.remove();
-    
-          // If any of the nodes were the start node, reset the start node
-          if (StateManager.startNode === node) {
-            StateManager.startNode = null;
+    /** Copies the selected objects, including any transitions between selected nodes. */
+    public static copySelectedObjects() {
+    // Copy selected nodes
+    const selectedNodes = this._selectedObjects.filter(obj => obj instanceof NodeWrapper) as NodeWrapper[];
```

Comment:
The indentation here looks like it may be incorrect - the function's body is indented to the same level as the function definition, I think?